allennlp/predictors/bidaf.py

```diff
@@ -38,3 +43,96 @@ def _json_to_instance(self, json_dict: JsonDict) -> Instance:
         question_text = json_dict["question"]
         passage_text = json_dict["passage"]
         return self._dataset_reader.text_to_instance(question_text, passage_text)
+
+    @overrides
+    def predictions_to_labeled_instances(self,
+                                         instance: Instance,
+                                         outputs: Dict[str, np.ndarray]) -> List[Instance]:
+        """
+        NAQANET has the following fields, answer_as_passage_spans,  answer_as_question_spans,
+        answer_as_add_sub_expressions, answer_as_counts. We need to provide labels for all of them.
+        """
+
+        # For BiDAF
+        if 'best_span' in outputs:
+            span_start_label = outputs['best_span'][0]
+            span_end_label = outputs['best_span'][1]
+            instance.add_field('span_start', IndexField(int(span_start_label), instance['passage'])) # type: ignore
+            instance.add_field('span_end', IndexField(int(span_end_label), instance['passage'])) # type: ignore
+
+        # For NAQANet model
+        elif 'answer' in outputs:
+            answer_type = outputs['answer']['answer_type']
+
+            # When the problem is a counting problem
+            if answer_type == 'count':
+                field = ListField([LabelField(int(outputs['answer']['count']), skip_indexing=True)])
+                instance.add_field('answer_as_counts', field)
+
+            # When the answer is in the passage
+            elif answer_type == 'passage_span':
+                span = outputs['answer']['spans'][0]
+
+                # Convert character span indices into word span indices
+                word_span_start = None
+                word_span_end = None
+                metadata_field = MetadataField(instance['metadata'])
```

Comment:
I don't understand why you're using `MetadataField` here, it seems like the only thing you do with it is pull the `instance['metadata']` back out of it?

that said, the `mypy` error is caused by your doing (in essence) 

```
for idx, offset in enumerate(instance['metadata']['passage_token_offsets']): ...
```

since mypy has no idea what it is you're enumerating. 

Unfortunately, there's not a great way to solve this, one bad way would be to use `typing.cast`, something like

```
passage_token_offsets = cast(List[Tuple[int, int]], instance['metadata']['passage_token_offsets'])
```

should work, or you could stick a `type: ignore` on the line

Reply:
was using Metadata field to try to get around the original mypy error.